internal/pkg/service/common/etcdop/prefix_test.go

```diff
@@ -157,35 +248,135 @@ func BenchmarkPrefix_GetAll(b *testing.B) {
 	if err := Key("key0").Put("foo").Do(ctx, etcd); err != nil {
 		b.Fatalf("cannot create etcd key: %s", err)
 	}
-	if err := Key("my/prefix/key1").Put("bar").Do(ctx, etcd); err != nil {
+	for i := 0; i < 100; i++ {
+		key := fmt.Sprintf("key%04d", i)
+		if err := pfx.Key(key).Put("bar").Do(ctx, etcd); err != nil {
+			b.Fatalf(`cannot create etcd key "%s": %s`, key, err)
+		}
+	}
+
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		kvs, err := pfx.GetAll().Do(ctx, etcd)
+		if err != nil || len(kvs) != 100 {
+			b.Fatalf("unexpected result")
+		}
+	}
+}
+
+func BenchmarkPrefix_DeleteAll(b *testing.B) {
+	ctx := context.Background()
+	etcd := etcdhelper.ClientForTest(b)
+
+	pfx := Prefix("my/prefix/")
+	if err := Key("key0").Put("foo").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+	if err := pfx.Key("key1").Put("bar").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+	if err := pfx.Key("key2").Put("baz").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		deleted, err := pfx.DeleteAll().Do(ctx, etcd)
+		if err != nil || (i == 0 != (deleted == 2)) { // xor
+			b.Fatalf("unexpected result")
+		}
+	}
+}
+
+func BenchmarkPrefixT_AtLestOneExists(b *testing.B) {
+	ctx := context.Background()
+	etcd := etcdhelper.ClientForTest(b)
+
+	pfx := typedPrefixForTest()
+	if err := Key("key0").Put("foo").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+	if err := pfx.Key("key1").Put("bar").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+	if err := pfx.Key("key2").Put("baz").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		found, err := pfx.AtLeastOneExists().Do(ctx, etcd)
+		if err != nil || !found {
+			b.Fatalf("unexpected result")
+		}
+	}
+}
+
+func BenchmarkPrefixT_Count(b *testing.B) {
+	ctx := context.Background()
+	etcd := etcdhelper.ClientForTest(b)
+
+	pfx := typedPrefixForTest()
+	if err := Key("key0").Put("foo").Do(ctx, etcd); err != nil {
 		b.Fatalf("cannot create etcd key: %s", err)
 	}
-	if err := Key("my/prefix/key2").Put("baz").Do(ctx, etcd); err != nil {
+	if err := pfx.Key("key1").Put("bar").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+	if err := pfx.Key("key2").Put("baz").Do(ctx, etcd); err != nil {
+		b.Fatalf("cannot create etcd key: %s", err)
+	}
+
+	b.StartTimer()
+
+	for i := 0; i < b.N; i++ {
+		count, err := pfx.Count().Do(ctx, etcd)
+		if err != nil || count != int64(2) {
+			b.Fatalf("unexpected result")
+		}
+	}
+}
+
+func BenchmarkPrefixT_GetAll(b *testing.B) {
+	ctx := context.Background()
+	etcd := etcdhelper.ClientForTest(b)
+
+	pfx := typedPrefixForTest()
+	if err := Key("key0").Put("foo").Do(ctx, etcd); err != nil {
 		b.Fatalf("cannot create etcd key: %s", err)
 	}
+	for i := 0; i < 100; i++ {
```

Comment:
Upravil som `GetAll` benchmarks, aby sa nacitavalo `100` klucov.

Pri `T` operaciach je nejaky narast v `memOps`, kedze sa tam robi aj validacia a kodovanie, ale nie velky.

Povodne som tam mal bug, a bolo toho `3x` viac, ale to som fixol (slice pre dekodovane hodnoty sa alokuje cela naraz).

![image](https://user-images.githubusercontent.com/19371734/204013392-10da955a-789f-4ae7-88ad-bd32e297a33e.png)